File: app/src/main/java/ma/aui/sse/csc3374/rpc/homework2/consumer/python/consumer.py
from zeep import Client
from pyfiglet import Figlet
from termcolor import colored
import base64

f = Figlet(font='slant')
serviceimpl = Client('src/main//resources/ServiceImplService.wsdl').service
def main_menu():
    while True:
        print(colored(f.renderText('Homework 2'), 'blue'))
        print(colored('Welcome to Homework 2', 'green'))
        print(colored('1.', 'yellow'), 'Get OsName.')
        print(colored('2.', 'yellow'), 'Get Screenshot.')
        print(colored('3.', 'yellow'), 'Reboot System.')
        print(colored('4.', 'yellow'), 'Exit.')
        choice = input(colored('\n--> Enter your Selection: ', 'blue'))
        if choice == '1':
            print(serviceimpl.getOsName()) # Function 1
        elif choice == '2': # Function 2
            base64url = serviceimpl.getScreenShot()
            outputpath = "C:\\Users\\yassi\\Downloads\\saved.png"
            with open(outputpath, 'wb') as fh:
                fh.write(base64.b64decode(base64url))
            print("Done.")
        elif choice == '3':
            serviceimpl.reboot() # Function 3
        elif choice =='4':
            return
        else:
            print(colored('--> Please select from the Menu.', 'red'))

main_menu()

File: app/src/main/java/ma/aui/sse/csc3374/rpc/homework2/consumer/js/consumer.js
let soap = require('soap');
global.atob = require("atob");

const chalk = require('chalk');
var fs = require('fs');
const prompt = require('prompt-sync')({sigint: true});
const path = require('path');
let url = 'http://localhost:9000/ServiceImpl?wsdl';

function getOS() {
    return new Promise((resolve, reject) => {
        soap.createClient(url, (err, client) => {
            client.getOsName((err,res) => {
                res ? resolve(res.return) : reject(err);
            })
        })
    })
}
function getScreenShot() {
    return new Promise((resolve, reject) => {
        soap.createClient(url, (err, client) => {
            client.getScreenShot((err,res) => {
                res ? resolve(res.return) : reject(err);
            })
        })
    })
}

function reboot() {
    return new Promise((resolve, reject) => {
        soap.createClient(url, (err, client) => {
            client.reboot((err,res) => {
                res ? resolve(res.return) : reject(err);
            })
        })
    })
}

async function menu() {
    
    let OsName, ScreenShotPath;
    
   
    while (true) {
    console.log("\n1. GET OsName\n2. SAVE ScreenShot\n3. REBOOT\n4. Exit\nChoice: ");
    let choice = prompt()

    if (parseInt(choice) === 1) {

        OsName = await getOS();
        console.log(chalk.red('OS Name: '), chalk.red(OsName));

    } else if (parseInt(choice) === 2) {

        const SC = await getScreenShot();
        ScreenShotPath = __dirname + '\\screenshot.png';
        console.log(chalk.green("File was successfully saved in: "), chalk.green(ScreenShotPath));
        fs.writeFile(ScreenShotPath, global.atob(SC), 'binary', function(err) { 
            console.log("DONE"); 
        });

    } else if (parseInt(choice) === 3) {
        await reboot();
    } 
    else if (parseInt(choice) === 4) {
        return;
    }else {
        console.log("Please enter a valid option");
    }
}
}
menu()